G/gmsh/build_tarballs.jl

```diff
@@ -3,20 +3,26 @@
 using BinaryBuilder
 
 name = "gmsh"
-version = v"4.13.1"
+version = v"4.15.0"
 
 # Collection of sources required to build Gmsh
 sources = [
     ArchiveSource("https://gmsh.info/src/gmsh-$(version)-source.tgz",
-                  "77972145f431726026d50596a6a44fb3c1c95c21255218d66955806b86edbe8d"),
-]
+                  "abb2632715bd7d0130ded7144fd6263635cd7dea883b8df61ba4da58ce6a1dfe"),
+    DirectorySource("./bundled")
+    ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd ${WORKSPACE}/srcdir/gmsh-*
 if [[ "${target}" == *linux* ]] || [[ "${target}" == *freebsd* ]]; then
     OPENGL_FLAGS="-DOpenGL_GL_PREFERENCE=LEGACY"
 fi
+
+if [[ ${target} == *mingw* ]]; then
+    atomic_patch -p1 "${WORKSPACE}/srcdir/patches/CMakeLists.txt.patch"
+fi
```

Comment:
Since the patch touches something which is mingw-specific anyway, it doesn't hurt to apply the patch always and reduce the platform-specific code.

Reply:
Also, adding comments to explain the rationale of a change would be very welcome.